code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/dao/AuthDao.java

```diff
@@ -18,7 +22,29 @@
 
     Account createAccount(String name, String kind, String externalId, String externalType);
 
+    Account createProject(String name, String description);
+
+    Account createDefaultProject(Account account);
+
     void updateAccount(Account account, String name, String kind, String externalId, String externalType);
 
-    List<? extends Account> getAccessibleProjects(String userAccountId, List<String> orgIds, List<String> teamIds);
+    List<Account> getAccessibleProjects(Set<ExternalId> externalIdSet, boolean isAdmin, Long policyAccount);
+
+    boolean hasAccessToProject(long projectId , Long policyAccount, boolean isAdmin, Set<ExternalId> externalIdSet);
```

Comment:
James, as you mentioned we shouldn't bind Dao methods with Policy, can we rename policyAccount to something else?